src/platform/telink/wifi/3.3.99.0/WiFiManager.cpp

```diff
@@ -179,6 +179,20 @@ CHIP_ERROR WiFiManager::Scan(const ByteSpan & ssid, ScanResultCallback resultCal
     mWiFiState          = WIFI_STATE_SCANNING;
     mSsidFound          = false;
 
+    if (!ssid.empty()) // Directed Scanning, Main Spec, 11.9.7.1. "ScanNetworks Command"
+    {
+        if (!mInternalScan) // don't erase in the middle of Connect procedure
+        {
+            Instance().mDirectedScanning = true;
+            mWantedNetwork.Erase();
+            memcpy(mWantedNetwork.ssid, ssid.data(), ssid.size());
+            mWantedNetwork.ssidLen = ssid.size();
+            ChipLogProgress(DeviceLayer, "Directed Scanning, looking for: %.*s",
+                            static_cast<int>(Instance().mWantedNetwork.GetSsidSpan().size()),
+                            Instance().mWantedNetwork.GetSsidSpan().data());
```

Comment:
```suggestion
                            static_cast<int>(ssid.size()), ssid.data());
```

This seems shorter ... or do we need the `Instance()` calls and `GetSsidSpan()` calls? Overall it seems awkward to use mWantedNetwork directly above and then go through `Instance()`